microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceASTValidator.java

```diff
@@ -276,34 +274,43 @@ private void validateRetryAnnotation(NormalAnnotation annotation)
 			Expression jitterUnitExpr = getAnnotationMemberValueExpression(
 					annotation, JITTER_DELAY_UNIT_RETRY_ANNOTATION_MEMBER);
 
-			int delayNum = (int) delayExpr.resolveConstantExpressionValue();
-			int maxDurationNum = (int) maxDurationExpr
-					.resolveConstantExpressionValue();
-			int jitterNum = jitterExpr != null
-					? (int) jitterExpr.resolveConstantExpressionValue()
-					: 0;
+			int delayNum = castMemberToInt(delayExpr);
+			int maxDurationNum = castMemberToInt(maxDurationExpr);
+			int jitterNum = castMemberToInt(jitterExpr);
 
-			Duration delayValue = findDurationUnit(delayUnitExpr, delayNum);
-			Duration maxDurationValue = findDurationUnit(durationUnitExpr,
-					maxDurationNum);
-			Duration jitterValue = findDurationUnit(jitterUnitExpr, jitterNum);
-
-			if (delayValue == null || maxDurationValue == null
-					|| jitterValue == null) {
+			if (delayNum == -1 || maxDurationNum == -1) {
 				return;
 			}
 
-			Duration maxDelayValue = delayValue.plus(jitterValue);
+			double delayValue = findDurationUnit(delayUnitExpr, delayNum);
+			double maxDurationValue = findDurationUnit(durationUnitExpr,
+					maxDurationNum);
+			double jitterValue = findDurationUnit(jitterUnitExpr, jitterNum);
+
+			double maxDelayValue = delayValue + jitterValue;
 
-			if (maxDelayValue.compareTo(maxDurationValue) >= 0) {
+			if (delayExpr != null && maxDurationExpr != null
+					&& maxDelayValue >= maxDurationValue) {
 				super.addDiagnostic(RETRY_WARNING_MESSAGE, DIAGNOSTIC_SOURCE,
 						delayExpr, DELAY_EXCEEDS_MAX_DURATION,
 						DiagnosticSeverity.Warning);
 			}
 		}
 	}
 
-	private Duration findDurationUnit(Expression memberUnitExpr,
+	private int castMemberToInt(Expression expr) {
+		if (expr != null) {
+			try {
+				return (int) expr.resolveConstantExpressionValue();

```

Comment:
For me, `expr.resolveConstantExpressionValue()` returns null, and the cast to int is what throws the NPE. To avoid that just get the result of `expr.resolveConstantExpressionValue()` and check whether its null. If it is, for any of delay, maxDuratio or jitter values, then we can avoid emitting a diagnostic since we can't cast the value to an integer. Most likely JDT will handle showing the correct diagnostic.

Reply:
I believe it's okay if jitter is null, delay and maxDuration should be checked for null.

Reply:
Ok, agreed.